spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -339,19 +342,25 @@ private Result doExecuteWithPartialProgress(RewriteExecutionContext ctx, Stream<
 
   private Stream<RewriteFileGroup> toGroupStream(RewriteExecutionContext ctx,
                                                  Map<StructLike, List<List<FileScanTask>>> fileGroupsByPartition) {
+    PriorityQueue<RewriteFileGroup> rewriteFileGroups = new PriorityQueue<>(comparator());
```

Comment:
Why this over say,

 Stream.sorted(comparator())?

Reply:
Thanks, I have made the changes.